packages/edit-navigation/src/components/menus-editor/create-menu-panel.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * External dependencies
+ */
+import { some } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	Button,
+	Panel,
+	PanelBody,
+	TextControl,
+	withFocusReturn,
+} from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useCallback, useEffect, useState } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+
+const noticeId = 'edit-navigation-create-menu-error';
+
+const menuNameMatches = ( menuName ) => ( menu ) =>
+	menu.name.toLowerCase() === menuName.toLowerCase();
+
+export function CreateMenuForm( { onCancel, menus } ) {
+	const [ menuName, setMenuName ] = useState( '' );
+	const { isCreatingMenu, hasCreatedMenu } = useSelect( ( select ) => {
+		const { getIsResolving, hasFinishedResolution } = select( 'core' );
+		return {
+			isCreatingMenu: getIsResolving( 'saveMenu' ),
+			hasCreatedMenu: hasFinishedResolution( 'saveMenu' ),
+		};
+	}, [] );
+	const { saveMenu } = useDispatch( 'core' );
+	const { createInfoNotice, createErrorNotice, removeNotice } = useDispatch(
+		'core/notices'
+	);
+
+	useEffect( () => {
+		if ( hasCreatedMenu ) {
+			createInfoNotice( __( 'Menu created' ), {
+				type: 'snackbar',
+				isDismissible: true,
+			} );
+		}
+	}, [ hasCreatedMenu ] );
+
+	const onCreateMenu = useCallback(
+		( event ) => {
+			// Prevent form submission.
+			event.preventDefault();
+
+			// Remove existing notices.
+			removeNotice( noticeId );
+
+			if ( menuName.length === 0 ) {
+				// Button is aria-disabled, do nothing.
+				return;
+			}
+
+			// Validate the menu name doesn't match an existing menu.
+			if ( some( menus, menuNameMatches( menuName ) ) ) {
+				const message = sprintf(
+					// translators: %s: the name of a menu.
+					__(
+						'The menu name %s conflicts with another menu name. Please try another.'
+					),
+					menuName
+				);
+				createErrorNotice( message, { id: noticeId } );
+				return;
+			}
+
+			saveMenu( { name: menuName } );
```

Comment:
Something like this should take care of the notifications.
```suggestion
			try {
				await saveMenu( { name: menuName } );
				createInfoNotice( __( 'Menu created' ), {
					type: 'snackbar',
					isDismissible: true,
				} );
			} catch (err) {
				createErrorNotice( __( 'Menu creation failed' ), {
					type: 'snackbar',
					isDismissible: true,
				} );
			}
```

Reply:
Thanks, that works nicely 👍